api/lcp/collection.py

```diff
@@ -0,0 +1,311 @@
+
+
+class LCPFulfilmentInfo(FulfillmentInfo):
+    """Sends LCP licenses as fulfilment info"""
+
+    def __init__(
+            self,
+            identifier,
+            collection,
+            data_source_name,
+            identifier_type,
+            content_link=None,
+            content_type=None,
+            content=None,
+            content_expires=None):
+        """Initializes a new instance of LCPFulfilmentInfo class
+
+        :param identifier: Identifier
+        :type identifier: string
+
+        :param collection: Collection
+        :type collection: Collection
+
+        :param data_source_name: Data source's name
+        :type data_source_name: string
+
+        :param identifier_type: Identifier's type
+        :type identifier_type: string
+
+        :param content_link: Content link
+        :type content_link: Optional[string]
+
+        :param content_link: Identifier's type
+        :type content_link: string
+
+        :param content: Identifier's type
+        :type content: Any
+
+        :param content_expires: Time when the content expires
+        :type content_expires: Optional[datetime.datetime]
+        """
+        super(LCPFulfilmentInfo, self).__init__(
+            collection,
+            data_source_name,
+            identifier_type,
+            identifier,
+            content_link,
+            content_type,
+            content,
+            content_expires
+        )
+
+    @property
+    def as_response(self):
+        """Returns LCP license as a Flask response
+
+        :return: LCP license as a Flask response
+        :rtype: Response
+        """
+        return send_file(
+            BytesIO(json.dumps(self.content)),
+            mimetype=DeliveryMechanism.LCP_DRM,
+            as_attachment=True,
+            attachment_filename='{0}.lcpl'.format(self.identifier)
+        )
+
+
+class LCPAPI(BaseCirculationAPI, ExternalIntegrationOwner):
+    """Implements LCP workflow"""
+
+    NAME = ExternalIntegration.LCP
+    SERVICE_NAME = 'LCP'
+    DESCRIPTION = 'Manually imported collection protected using Readium LCP DRM'
+
+    SETTINGS = [
+        LCPEncryptionConfiguration.lcpencrypt_location.to_settings(),
+        LCPEncryptionConfiguration.lcpencrypt_output_directory.to_settings(),
+        LCPServerConfiguration.lcpserver_url.to_settings(),
+        LCPServerConfiguration.lcpserver_user.to_settings(),
+        LCPServerConfiguration.lcpserver_password.to_settings(),
+        LCPServerConfiguration.lcpserver_input_directory.to_settings(),
+        LCPServerConfiguration.lcpserver_page_size.to_settings(),
+        LCPServerConfiguration.provider_name.to_settings(),
+        LCPServerConfiguration.passphrase_hint.to_settings(),
+        LCPServerConfiguration.encryption_algorithm.to_settings(),
+        LCPServerConfiguration.max_printable_pages.to_settings(),
+        LCPServerConfiguration.max_copiable_pages.to_settings()
+    ]
```

Comment:
As I mentioned in the core branch, I would expect this to be more like 

```
SETTINGS = [
 [member.to_settings() for name, member in ConfigurationMetadata.get_configuration_metadata(LCPEncryptionConfiguration)]
+
[member.to_settings() for name, member in ConfigurationMetadata.get_configuration_metadata(LCPServerConfiguration)]
]
```

Or something even simpler with an additional helper function.

Reply:
Changed it to `SETTINGS = LCPServerConfiguration.to_settings() + LCPEncryptionConfiguration.to_settings()`